flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/internal/BatchTableEnvImpl.scala

```diff
@@ -133,6 +135,26 @@ abstract class BatchTableEnvImpl(
     }
   }
 
+  /**
+    * Key by the partition fields if the sink is a [[PartitionableTableSink]].
+    * @param sink    the table sink
+    * @param dataSet the data set
+    * @tparam R      the data set record type
+    * @return a data set that maybe keyed by.
+    */
+  private def mayKeyByPartitionFields[R](
```

Comment:
What about `applyShuffleByPartitionFieldsIfNeeded` ?

Reply:
Renamed to shuffleByPartitionFieldsIfNeeded.